### PR TITLE
[prometheus-postgres-exporter] Add support for DATA_SOURCE_USER_FILE

### LINKED ISSUE #4719: [prometheus-postgres-exporter] Add support for `DATA_SOURCE_USER_FILE` environmental variable
### Is your feature request related to a problem ?

We use Vault dynamic database secrets, where both username and password are dynamically generated. We then use a Vault injector to inject a username and password into a pod as a file. I see that Postgres exporter supports the variable `DATA_SOURCE_USER_FILE`, where you specify the name of the file where the username is rendered. This is currently not supported in config: https://github.com/prometheus-community/helm-charts/blob/62734a52836589518ca1921224ccbe6c975bdb87/charts/prometheus-postgres-exporter/values.yaml#L126-L155

### Describe the solution you'd like.

I want to specify the filename in a configuration. This is how the configuration should look like:
```
config:
  datasource:
    usernameFile: "/path/to/my/user/file"
```

The template would look similar to this: https://github.com/prometheus-community/helm-charts/blob/62734a52836589518ca1921224ccbe6c975bdb87/charts/prometheus-postgres-exporter/templates/deployment.yaml#L103-L115

```
          {{- if .Values.config.datasource.usernameFile }}
          - name: DATA_SOURCE_USER_FILE
            value: {{ tpl .Values.config.datasource.usernameFile . }}
          {{- else }}
          - name: DATA_SOURCE_USER
            valueFrom:
              secretKeyRef:
          {{- if .Values.config.datasource.usernameSecret }}
                name: {{ tpl .Values.config.datasource.usernameSecret.name . }}
                key: {{ tpl .Values.config.datasource.usernameSecret.key . }}
          {{- else }}
                name: {{ template "prometheus-postgres-exporter.fullname" . }}
                key: data_source_password
```

### Describe alternatives you've considered.

I can manually set the environmental variable:
```
extraEnvs:
  - name: "DATA_SOURCE_USER_FILE"
    value: "/path/to/my/user/file"
```

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by ubajze**
#### What this PR does / why we need it
This PR adds support for the `DATA_SOURCE_USER_FILE` variable for the postgres-exporter helm charts.

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #4719

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by zeritti — CHANGES_REQUESTED** [review]
Thank you, @ubajze, for your PR. Please, see my comments below. Remember to bump the chart version number (this would be a minor change).

**Comment by ubajze** [comment]
@zeritti Thank you for the suggestions. I also bumped the chart version.

File: charts/prometheus-postgres-exporter/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 appVersion: "v0.15.0"
 description: A Helm chart for prometheus postgres-exporter
 name: prometheus-postgres-exporter
-version: 6.0.1
+version: 6.0.2
```
Comment (zeritti) [inline charts/prometheus-postgres-exporter/Chart.yaml:5]: We have to consider this change a minor one as a new feature has ben introduced.
```suggestion
version: 6.1.0
```

**Review by zeritti — CHANGES_REQUESTED** [review]
Thank you, @ubajze, for your changes. The chart version still needs a correction.

File: charts/prometheus-postgres-exporter/Chart.yaml (hunk 1 above)
Comment (ubajze) [reply charts/prometheus-postgres-exporter/Chart.yaml:5]: Ah yeah, you said minor, not patch. Sorry, my bad. Committing the change.

**Review by zeritti — APPROVED** [review]
Thank you, @ubajze, LGTM.

**Pull request merged**